### ISSUE TITLE
travis checks should ensure that the idranges file is valid

### ISSUE THREAD

**Issue by cmungall** [INCATools/ontology-development-kit#211]
Currently idranges.owl is never parsed by machine, it is informational for humans to adjust their Protege settings. We should have more support for this, and it is likely Protege will include support in the future (thanks @matthewhorridge!)

Basic validation: Should be valid OWL, and id ranges should not overlap. 

Note that GO's idranges is invalid right now as the ranges include the prefix - it has never been parsed!

**Comment by matthewhorridge** [comment]
In 5.6.0 Protege will look for -idranges.owl files and parse these.  After loading the "New Entity" settings will automatically be updated.  The id numeric range will be set if the current user name is found in the list of id ranges.  If it isn't then a dialog will be displayed...

<img width="698" alt="image" src="https://user-images.githubusercontent.com/488607/56780911-f59ce900-6795-11e9-8690-2966a8107e5c.png">

It might be a good idea to use Git user names in the idranges OWL file and set Protege to use the current git user name (for ontologies loaded from git repos).

**Comment by cmungall** [comment]
This is great. 

We could either just have the convention that allocatedto is the github name, OR we can add a new property (I would favour using a property from doap rather than minting in IAO)

There may be scenarios where one person has multiple ranges, for different "roles", or sometimes templates may be subdivided into ranges. Thoughts @dosumis @matentzn @nicolevasilevsky @pbuttigieg ?

**Comment by matthewhorridge** [comment]
> There may be scenarios where one person has multiple ranges, for different "roles", or sometimes templates may be subdivided into ranges. 

Yes, I've seen an example like this "Matthew (home)", "Matthew (office)", so having multiple aliases would be good IMO.

**Comment by pbuttigieg** [comment]
+1 for this functionality. Could this be bundled with autodetecting the PURL format and namespace? Seems that the id-ranges file is a good home for this information

Using GitHub user names sounds good, as this can be used to autotag editors. I'd also have fields for full name and ORCID.

**Comment by gouttegd** [comment]
We already have a `validate_idranges` target in the Makefile, all that is needed is to call that target when running the CI checks.
